### PR TITLE
Fix DataCache.Find

### PR DESCRIPTION
**Pull request by Tommo-L**
Closes https://github.com/neo-project/neo/issues/1672

### CODE REVIEW

**Comment by roman-khimov** [comment]
What if instead
```
             myDataCache.TryGet(new MyKey("key3")); // GETLINE
```
I'd do an equivalent of `Storage.Put("key3", "value33")`? Which value will I get for this key (especially if I'm to `myDataCache.Add(new MyKey("key3"), new MyValue("value3"));` in setup phase)? [sorry, I don't have 3.0-compatible .net installation at hand to check]

**Comment by erikzhang** [comment]
> Which value will I get for this key

You will get the old value.

File: src/neo/IO/Caching/DataCache.cs [hunk 1]
```diff
@@ -165,7 +165,7 @@ public IEnumerable<(TKey Key, TValue Value)> Find(byte[] key_prefix = null)
                     ))
                     .OrderBy(p => p.KeyBytes, ByteArrayComparer.Default)
                     .ToArray();
-                cachedKeySet = dictionary.Select(p => p.Key).ToHashSet();
+                cachedKeySet = new HashSet<TKey>(cached.Select(p => p.Item2));
```
Comment (Tommo-L) [inline src/neo/IO/Caching/DataCache.cs:168]: This will lost `TrackState.Deleted` items?

Comment (Tommo-L) [reply src/neo/IO/Caching/DataCache.cs:168]: Maybe we can use `new HashSet<TKey>(cached.Select(p => p.Item2).concat(changeSet))`.

Comment (erikzhang) [reply src/neo/IO/Caching/DataCache.cs]: Fixed.

**Pull request merged**